configure.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/sh
+
+die()
+{
+    echo "$0: $@" 1>&2
+    exit 1
+}
+
+cc_is_clang()
+{
+    ${CC:-cc} -v 2>&1 | grep -q "clang version"
+}
+
+cc_is_gcc()
+{
+    ${CC:-cc} -v 2>&1 | grep -q "^gcc version"
+}
+
+check_ld_version()
+{
+    type $1 >/dev/null 2>&1 && $1 -v | grep -q 'GNU.*2.2[5-9]'
+}
+
+# Host-provided header files are installed here for in-tree builds. OPAM will
+# install these to $(OPAM_INCDIR)/host where they will be picked up by
+# pkg-config.
+HOST_INCDIR=${PWD}/include-host
+
+case $(uname -s) in
+    Linux)
+        # On Linux/gcc we use -nostdinc and copy all the gcc-provided headers.
+        cc_is_gcc || die "Only 'gcc' is supported on Linux"
+        CC_INCDIR=$(${CC:-cc} -print-file-name=include)
+        cp -R ${CC_INCDIR} ${HOST_INCDIR}
+
+        HOST_CFLAGS="-nostdinc"
+        BUILD_UKVM="yes"
+        BUILD_VIRTIO="yes"
+        ;;
+    FreeBSD)
+        # On FreeBSD/clang we use -nostdlibinc which gives us access to the
+        # clang-provided headers for compiler instrinsics. We copy the rest
+        # (std*.h, float.h) from the host.
+        cc_is_clang || die "Only 'clang' is supported on FreeBSD"
+        INCDIR=/usr/include
+        SRCS_MACH="machine/_stdint.h machine/_types.h machine/endian.h \
+            machine/_limits.h"
+        SRCS_SYS="sys/_null.h sys/_stdint.h sys/_types.h sys/cdefs.h"
```

Comment:
nocrypto (or zarith or gmp or whatever other dependency) also needs `sys/endian.h` (sorry that my blog article is out of date)